### PR TITLE
Add tests with Rspec

### PR DESCRIPTION
**Pull request by danblitzhou**
Fixing: #44 

Run `rspec` to execute all tests

Model tests:

- [x] AdminUser
- [x] Post
- [x] All post types

Controller tests:
sdfsfsfs.....

### CODE REVIEW

File: app/models/admin_user.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@ def self.root_user
   protected
 
   def not_adding_root_user
-    is_root_user = AdminUser.root_user.id == id
+    is_root_user = AdminUser.root_user&.id == id
```
Comment (Liang-Hsuan) [inline app/models/admin_user.rb]: What's the benefit of adding `&`?

File: spec/support/factory_bot.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+RSpec.configure do |config|
+    config.include FactoryBot::Syntax::Methods
```
Comment (Liang-Hsuan) [inline spec/support/factory_bot.rb]: So `FactoryBot` is a dependency for `RSpec`?

File: spec/models/admin_user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+require "rails_helper"
+
+RSpec.describe AdminUser, :type => :model do
+  context "with root as email prefix" do
```
Comment (Liang-Hsuan) [inline spec/models/admin_user_spec.rb]: Nice, correct way of using `context` and `it`.

File: spec/factories/admin_user.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+FactoryBot.define do
+    factory :admin_user do
+      email "user@kicks4love.com"
+      username "Robin Lou"
+      password "grandcherokee"
```
Comment (Liang-Hsuan) [inline spec/factories/admin_user.rb]: I need some explanation on what is going on here 😅

File: spec/support/factory_bot.rb (hunk 2 above)
Comment (danblitzhou) [reply spec/support/factory_bot.rb]: not really, it just makes thing easier for me

File: spec/factories/admin_user.rb (hunk 4 above)
Comment (danblitzhou) [reply spec/factories/admin_user.rb]: 🐴

File: app/models/admin_user.rb (hunk 1 above)
Comment (danblitzhou) [reply app/models/admin_user.rb]: in case there is no AdminUser in the first place

File: spec/support/factory_bot.rb (hunk 2 above)
Comment (Liang-Hsuan) [reply spec/support/factory_bot.rb]: Ok, will discuss it offline, I need more details on this

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+          name: install and setup elasticsearch
+          command: |
+            wget -qO - https://artifacts.elastic.co/GPG-KEY-elasticsearch | sudo apt-key add -
+            sudo apt-get install apt-transport-https
+            echo "deb https://artifacts.elastic.co/packages/6.x/apt stable main" | sudo tee -a /etc/apt/sources.list.d/elastic-6.x.list
+            sudo apt-get update && sudo apt-get install elasticsearch
+
+      # Database setup
+      - run: mv config/database.ci.yml config/database.yml
+      - run: mv config/secrets.ci.yml config/secrets.yml
```
Comment (Liang-Hsuan) [inline .circleci/config.yml]: Also let's see if only use `SECRET_KEY_BASE` env var works

**Pull request merged**